### PR TITLE
Add an option erlang-icr-indent

### PR DESCRIPTION
**Pull request by victorhge**
The option makes it possible to customize the indention of if/case/recieve patterns.

Before the change, the indentaion of if/case/recieve patterns is
4 steps and it cannot be changed alone.  Some people prefer other
styles, for example, zero indentation.

case a of
true ->
   do_something();
false ->
   do_something_else()
end.

This patch just adds an option, which default value is 4 steps.
"(setq erlang-icr-indent 0)" will keep if/case/receive at the same column of
the sub-clauses.

### CODE REVIEW

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by dgud** [comment]
Can you rebase this to maint.

**Comment by victorhge** [comment]
Why?  And what do you mean by "this"?   The pull request?  How to do that?

**Comment by dgud** [comment]
Why?  I wanted to include this PR in the next (maint) release
This? your branch and the PR is based on the master branch.
How? See man page for git rebase --onto

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by victorhge** [comment]
Now the branch and PR are based on maint branch.

**Comment by dgud** [comment]
Can you update you maint branch and rebase your branch on top of that, and solve the merge conflicts.

**Comment by victorhge** [comment]
Conflicts solved.

**Comment by dgud** [comment]
Sorry I should have written this in the previous comment:
And squash it to one commit.

File: lib/tools/emacs/erlang.el [hunk 1]
```diff
@@ -2948,7 +2950,7 @@ Return nil if inside string, t if in a comment."
                       ;; If in fun definition use standard indent level not double
                       ;;(if (not (eq (car (car (cdr stack))) 'fun))
                       ;; Removed it made multi clause fun's look to bad
-                      (setq off (* 2 erlang-indent-level)))) ;; )
+                      (setq off (+ erlang-indent-levelerlang-icr-indent))))
```
Comment (dgud) [inline lib/tools/emacs/erlang.el]: A space was lost here.

**Comment by victorhge** [comment]
I thought the default indentation was always 4.  It turns out `if' options is 3. 
I have changed the default value of erlang-icr-indent to `nil',  which means keep the default behavior.  When non-nil,  the pattens of if/case/receive clause are indented according to erlang-icr-indent.

**Pull request merged**